src/fu-quirks.h

```diff
@@ -22,18 +23,112 @@
 #ifndef __FU_QUIRKS_H
 #define __FU_QUIRKS_H
 
-typedef struct {
-	const gchar             *sys_vendor;
-	const gchar		*identifier;
-	AsVersionParseFlag       flags;
-} FuVendorQuirks;
+G_BEGIN_DECLS
 
-static const FuVendorQuirks quirk_table[] = {
-	/* Dell & Alienware use AA.BB.CC.DD rather than AA.BB.CCDD */
-	{ "Dell Inc.",	"com.dell.uefi",	AS_VERSION_PARSE_FLAG_NONE },
-	{ "Alienware",	"com.dell.uefi",	AS_VERSION_PARSE_FLAG_NONE },
-	{ NULL,		NULL,			AS_VERSION_PARSE_FLAG_NONE }
-};
+#include <glib-object.h>
+#include <gusb.h>
 
-#endif /* __FU_QUIRKS_H */
+#define FU_TYPE_QUIRKS (fu_quirks_get_type ())
+G_DECLARE_FINAL_TYPE (FuQuirks, fu_quirks, FU, QUIRKS, GObject)
+
+FuQuirks	*fu_quirks_new				(void);
+gboolean	 fu_quirks_load				(FuQuirks	*self,
+							 GError		**error);
+const gchar	*fu_quirks_lookup_by_id			(FuQuirks	*self,
+							 const gchar	*prefix,
+							 const gchar	*id);
+const gchar	*fu_quirks_lookup_by_glob		(FuQuirks	*self,
+							 const gchar	*prefix,
+							 const gchar	*glob);
+const gchar	*fu_quirks_lookup_by_usb_device		(FuQuirks	*self,
+							 const gchar	*prefix,
+							 GUsbDevice	*dev);
+
+/**
+ * FU_QUIRKS_PLUGIN_DFU:
+ * @key: the USB device ID, e.g. `USB\VID_0763&PID_2806`
+ * @value: a string, separated using `|`, e.g. `ignore-polltimeout|no-pid-change`
+ *
+ * Assigns optional quirks to use for a DFU device which does not follow the
+ * DFU 1.0 or 1.1 specification.
+ *
+ * Default value: `none`
+ *
+ * Since: 1.0.1
+ */
+#define	FU_QUIRKS_PLUGIN_DFU			"fwupd-plugin-dfu"
+
+/**
+ * FU_QUIRKS_PLUGIN_UEFI_VERSION_FORMAT:
+ * @key: a %FU_HWIDS_KEY_MANUFACTURER, e.g. `Alienware`
+ * @value: the version format, e.g. `none`
+ *
+ * Assigns the version format to use for a specific manufacturer.
+ * A specific version format is sometimes chosen to match the appearance of
+ * other systems or specifications.
+ *
+ * Default value: `use-triplet`
+ *
+ * Since: 1.0.1
+ */
+#define	FU_QUIRKS_PLUGIN_UEFI_VERSION_FORMAT	"fwupd-plugin-uefi-version-format"
+
+/**
+ * FU_QUIRKS_DAEMON_VERSION_FORMAT:
+ * @key: the optionally wildcarded AppStream ID e.g. `com.dell.uefi*.firmware`
+ * @value: the version format, e.g. `none`
+ *
+ * Assigns the version format to use for a specific AppStream component.
+ * A specific version format is sometimes chosen to match the appearance of
+ * other systems or specifications.
+ *
+ * Default value: `use-triplet`
+ *
+ * Since: 1.0.1
+ */
+#define	FU_QUIRKS_DAEMON_VERSION_FORMAT		"fwupd-daemon-version-format"
```

Comment:
I think put the UEFI plugin as the last plugin, and the daemon at the very end.  So all plugin quirks come in order (alphabetically) and then daemon quirks come afterwards.   Makes them easier to find rather  UEFI in between all the DFU ones.

Reply:
so `fwupd-dfu-alternate-vidpid-plugin`, `fwupd-dfu-jabra-detach-plugin`, `fwupd-uefi-version-format-plugin` and `fwupd-version-format-daemon`? Doing it this way moves the "where" to right at the end. Maybe we drop the `-plugin` suffix entirely?

Reply:
I think drop the suffix entirely yes.  Order sounds right.